### PR TITLE
Support Table Partitions

### PR DESCRIPTION
**Pull request by avivl**

### CODE REVIEW

File: src/bigquery_query.ts [hunk 1]
```diff
@@ -338,7 +344,7 @@ export default class BigQueryQuery {
         return q.replace(/\$__timeFilter\(([\w_.]+)\)/g, range);
     }
 
-    static _getInterval(q, alias) {
+    static _getInterval(q, alias: boolean) {
```
Comment (liorama) [inline src/bigquery_query.ts]: `q` string?

File: src/query_ctrl.ts [hunk 2]
```diff
@@ -217,14 +218,19 @@ export class BigQueryQueryCtrl extends QueryCtrl {
...
         this.target.sharded = false;
+        this.target.partitioned = false;
         this.target.table = this.tableSegment.value;
         this.applySegment(this.timeColumnSegment, this.fakeSegment('-- time --'));
         let sharded = this.target.table.indexOf("_YYYYMMDD");
         if (sharded > -1) {
             this.target.table = this.target.table.substring(0, sharded + 1) + "*";
             this.target.sharded = true;
+       }
+        let partitioned = this.target.table.indexOf("__partitioned");
```
Comment (liorama) [inline src/query_ctrl.ts]: Use ```const``` and if you can bring this declaration to the top of the scope.

File: src/bigquery_query.ts [hunk 3]
```diff
@@ -246,6 +246,12 @@ export default class BigQueryQuery {
             let to = BigQueryQuery.formatDateToString(this.templateSrv.timeRange.to._d);
             query += " AND  _TABLE_SUFFIX BETWEEN \'" + from +  "\' AND \'" + to + "\' ";
         }
+        if (this.target.partitioned) {
+            console.log(BigQueryQuery.formatDateToString(this.templateSrv.timeRange.to._d,'-',  true))
+            query += " AND _PARTITIONTIME >= \'" +
```
Comment (liorama) [inline src/bigquery_query.ts]: Use string literals for templating.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Comment (avivl) [reply src/bigquery_query.ts]: @liorama thanks! still working on my typescript skills. Will add more type checks as I go

**Pull request merged**